src/commands/basemaps-mapsheet/create-mapsheet.ts

```diff
@@ -77,6 +81,19 @@ export const basemapsCreateMapSheet = command({
   },
 });
 
+/**
+ * Read Basemaps bundled config file by a given path can be both local or s3 path. Also support for gunzip json file.
+ *
+ */
+async function readConfig(config: string): Promise<ConfigBundled> {
+  const data = await fsa.read(config);
+  if (config.endsWith('.gz')) {
+    return JSON.parse((await pGunzip(data)).toString()) as ConfigBundled;
```

Comment:
Is `promisify` + `await` a common pattern? I've not seen that one before. Would it be easier to do `return gunzip(data, (error, result) => JSON.parse(result))`?

Reply:
@blacha Adding this feature into [@chunkd](https://github.com/blacha/chunkd/pull/1345/files), I think we will close this and just upgrade to use the new @chunkd later.

Reply:
Javascript is a async language, so most things that have a `callback` don't actually return anything

```typescript 
const x = gunzip(data, (error, result) => JSON.parse(result))
console.log(typeof x, x) // "undefined", undefined
```

`promisfy` is a function that converts the old callback syntax into something you can return and use in the new `async/await` syntax

```typescript
const y = await pGunzip(data)
console.log(y) // data
```

@Wentao-Kuang We need this pull request still as upgrading to latest chunkd will be too much work for this story.